HTTP Action docs update (#199)
Knotx/knotx-fragments: action/library/README.md

# Action Library
It is a core [actions](https://github.com/Knotx/knotx-fragments/tree/master/action#action) and [behaviours](https://github.com/Knotx/knotx-fragments/tree/master/action#behaviour) 
library.

## Action
It is a list of core actions:

> - [HTTP Action](#http-action) - invokes a web API and stores a response body in a fragment's payload

Reviewer: ```suggestion
- [HTTP Action](#http-action) - invokes a web API and stores a response body in Fragment's payload
```

> - [Inline Body Action](#inline-body-action) - replaces a fragment's body with the configured value

Reviewer: ```suggestion
- [Inline Body Action](#inline-body-action) - replaces Fragment's body with the configured value
```

> - [Inline Payload Action](#inline-payload-action) - adds a JSON data into a fragment's payload

Reviewer: ```suggestion
- [Inline Payload Action](#inline-payload-action) - adds a JSON data into Fragment's payload
```

- [CopyPayloadKeyActionFactory](#copy-payload-key-action) - copies data inside Fragment's payload

> - [Payload To Body Action](https://github.com/Knotx/knotx-fragments/tree/master/action/library#payload-to-body-action) - rewrites a fragment's payload to the body

Reviewer: ```suggestion
- [Payload To Body Action](https://github.com/Knotx/knotx-fragments/tree/master/action/library#payload-to-body-action) - rewrites Fragment's payload to its body
```

### HTTP Action
HTTP Action calls a Web API (e.g. RESTful API), decodes the response body (e.g. text to JSON object) and finally stores all response details in a fragment's 
payload. It [supports](#supported-http-methods) `GET`, `POST`, `PUT`, `PATCH`, `DELETE` and `HEAD` HTTP methods.

#### How does it work
It uses [Vert.x Web Client](https://vertx.io/docs/vertx-web-client/java/), an asynchronous HTTP and HTTP/2 client, to call web APIs and handle responses. All 
request/response interaction aspects are fully configurable:
- web API domain, port and path
- response body decoding (plain String, Json object or Json Array)
- success criteria (it uses [Vert.x Web Client response predicates](https://vertx.io/docs/vertx-web-client/java/#response-predicates))

The sample configuration for fetching resources (`/products`) from REST API:
```
fetch-products {
  factory = http
  config {
    httpMethod = GET # this is the default so can be skipped
    endpointOptions {
      path = /products
      domain = localhost
      port = 8080
      allowedRequestHeaders = ["Content-Type"]
    }
    responseOptions {
      predicates = [SC_SUCCESS, JSON]
      forceJson = false
    }
  }
}
```

> The action does the HTTP `GET` request to `http://localhost:8080/products`. By default, a Vert.x Web Client request ends with an error only if something wrong 

Reviewer: ```suggestion
The action does the HTTP `GET` request to `http://localhost:8080/products`. By default, Vert.x Web Client's request ends with an error only if something wrong 
```

happens at the network level. The response options (success conditions) indicate that we expect:
- `2xx` response status code
- `content-type` response header is `application.json`
- response body is a json object or json array

Please note that all client (`4xx`) and server (`5xx`) error response status codes fail the action.  

For the above example, when all processing ends with success, and the response body contains:
```
[
  {
    "id": "sku-01",
    "name": "Vert.x in Action"
  },
  {
    "id": "sku-02",
    "name": "Knot.x in Action"
  }
]
```

the action fragment result will contain:

> - transition is `_success`

Reviewer: ```suggestion
-  `_success` transition
```

```

## Behaviour
It is a list of core actions:

> - [Circuit Breaker Be](#circuit-breaker-behaviour) - wraps an action with circuit breaker

Reviewer: ```suggestion
- [Circuit Breaker Behaviour](#circuit-breaker-behaviour) - wraps an action with a circuit breaker
```
